packages/pwa-kit-react-sdk/src/ssr/universal/withReactQuery.js

```diff
@@ -0,0 +1,58 @@
+import hoistNonReactStatic from 'hoist-non-react-statics'
+import {FetchStrategy} from './fetchStrategy'
+import React from 'react'
+import {dehydrate, Hydrate, QueryClient, QueryClientProvider} from '@tanstack/react-query'
+import ssrPrepass from 'react-ssr-prepass'
+
+const isServerSide = typeof window === 'undefined'
+const STATE_KEY = '__reactQuery'
+
+export const withReactQuery = (Wrapped) => {
+    const wrappedComponentName = Wrapped.displayName || Wrapped.name
+
+    class WithReactQuery extends FetchStrategy {
+        constructor(props) {
+            super(props)
+
+            // Not crazy about this, but it's *super* important that
+            // we avoid making queryClient global – it can't be shared
+            // between requests.
+            if (!isServerSide && !this.props.locals.__queryClient) {
+                this.props.locals.__queryClient = new QueryClient()
+            }
+        }
+
+        render() {
+            return (
+                <QueryClientProvider client={this.props.locals.__queryClient}>
+                    <Hydrate state={isServerSide ? {} : window.__PRELOADED_STATE__?.[STATE_KEY]}>
+                        <Wrapped {...this.props} />
+                    </Hydrate>
+                </QueryClientProvider>
+            )
+        }
+
+        static async doInitAppState({res, appJSX}) {
+            const queryClient = (res.locals.__queryClient = new QueryClient())
```

Comment:
This is a little wonky innit? 

Right at this point in time we are changing the reference that the QueryClientProvider component is using in the JSX below, but only doing this on the server, since this function is only run on the server. 

I ran into a similar problem when I was working on this as well. The route I chose was to do the HOC enhancing at runtime. I'm assuming we have to do this here because we are using a standard method of using those HOC's at compile time (e.g. in your app config file default export)